diff --git a/aptos-node/src/utils.rs b/aptos-node/src/utils.rs
index 7ddfddf5ee..e331675a9c 100644
--- a/aptos-node/src/utils.rs
+++ b/aptos-node/src/utils.rs
@@ -54,7 +54,11 @@ pub fn set_aptos_vm_configurations(node_config: &NodeConfig) {
     let effective_concurrency_level = if node_config.execution.concurrency_level == 0 {
         min(
             DEFAULT_EXECUTION_CONCURRENCY_LEVEL,
-            (num_cpus::get() / 2) as u16,
+            if num_cpus::get() > 1 {
+                (num_cpus::get() / 2) as u16
+            } else {
+                1
+            },
         )
     } else {
         node_config.execution.concurrency_level
